logpuzzle.py

```diff
@@ -29,7 +32,12 @@ def read_urls(filename):
     Screens out duplicate urls and returns the urls sorted into
     increasing order."""
     # +++your code here+++
-    pass
+    head, _, tail = filename.partition("_")
+    filename_re = re.compile(r'GET\s/(\S*puzzle\S*)\s')
+    with open(filename, 'rt') as f_obj:
```

Comment:
FYI. The open built-in returns a file handler. By convention, the hander is labeled f or handler.